systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java

```diff
@@ -199,4 +207,35 @@ public static void replaceKafkaResource(String resourceName, Consumer<Kafka> edi
         ResourceManager.replaceCrdResource(Kafka.class, KafkaList.class, DoneableKafka.class, resourceName, editor);
     }
 
+    public static String getKafkaTlsListenerCaCertName(String namespace, String clusterName) {
+        KafkaListenerTls kafkaListenerTls = kafkaClient().inNamespace(namespace).withName(clusterName).get().getSpec().getKafka().getListeners().getTls();
+        return kafkaListenerTls.getConfiguration() == null ?
+                KafkaResources.clusterCaCertificateSecretName(clusterName) : kafkaListenerTls.getConfiguration().getBrokerCertChainAndKey().getSecretName();
+    }
+
+    public static String getKafkaExternalListenerCaCertName(String namespace, String clusterName) {
+        KafkaListenerExternal kafkaListenerExternal = kafkaClient().inNamespace(namespace).withName(clusterName).get().getSpec().getKafka().getListeners().getExternal();
+
+        KafkaListenerExternalConfiguration kafkaListenerExternalConfiguration = null;
+
+        switch (kafkaListenerExternal.getType()) {
+            case KafkaListenerExternalRoute.TYPE_ROUTE:
+                kafkaListenerExternalConfiguration = ((KafkaListenerExternalRoute) kafkaListenerExternal).getConfiguration();
+                break;
+            case KafkaListenerExternalNodePort.TYPE_NODEPORT:
+                kafkaListenerExternalConfiguration = ((KafkaListenerExternalNodePort) kafkaListenerExternal).getConfiguration();
+                break;
+            case KafkaListenerExternalLoadBalancer.TYPE_LOADBALANCER:
+                kafkaListenerExternalConfiguration = ((KafkaListenerExternalLoadBalancer) kafkaListenerExternal).getConfiguration();
+                break;
+            case KafkaListenerExternalIngress.TYPE_INGRESS:
+                kafkaListenerExternalConfiguration = ((KafkaListenerExternalIngress) kafkaListenerExternal).getConfiguration();
+                break;
+            default:
+                break;
```

Comment:
Can you specify in default new value for example like `KafkaListenerExternalIngress.TYPE_NONE`, basically use it instead of null when you comparing in return or it isn't possible?